qiskit_machine_learning/algorithms/inference/qbayesian.py

```diff
@@ -0,0 +1,335 @@
+                If ``None`` is given, a default instance of the reference sampler defined
+                by :class:`~qiskit.primitives.Sampler` will be used.
+        Raises:
+            ValueError: If any register in the circuit is not mapped to exactly one qubit.
+        """
+        # Test valid input
+        for qrg in circuit.qregs:
+            if qrg.size > 1:
+                raise ValueError("Every register needs to be mapped to exactly one unique qubit")
+        # Initialize parameter
+        self._circ = circuit
+        self.shots = shots
+        self.limit = limit
+        self.threshold = threshold
+        if sampler is None:
+            sampler = Sampler()
+        self.sampler = sampler
+
+        # Label of register mapped to its qubit
+        self._label2qubit = {qrg.name: qrg[0] for qrg in self._circ.qregs}
+        # Label of register mapped to its qubit index bottom up in significance
+        self._label2qidx = {
+            qrg.name: self._circ.num_qubits - idx - 1 for idx, qrg in enumerate(self._circ.qregs)
+        }
+        # Distribution of samples from rejection sampling
+        self.samples: Dict[str, float] = {}
+        # True if rejection sampling converged after limit
+        self.converged = bool()
+
+    def _get_grover_op(self, evidence: Dict[str, int]) -> GroverOperator:
+        """
+        Constructs a Grover operator based on the provided evidence. The evidence is used to
+        determine the "good states" that the Grover operator will amplify.
+
+        Args:
+            evidence: A dictionary representing the evidence with keys as variable labels
+                and values as states.
+        Returns:
+            GroverOperator: The constructed Grover operator.
+        """
+        # Evidence to reversed qubit index sorted by index
+        num_qubits = self._circ.num_qubits
+        e2idx = sorted(
+            [(self._label2qidx[e_key], e_val) for e_key, e_val in evidence.items()],
+            key=lambda x: x[0],
+        )
+        # Binary format of good states
+        num_evd = len(e2idx)
+        bin_str = [
+            format(i, f"0{(num_qubits - num_evd)}b") for i in range(2 ** (num_qubits - num_evd))
+        ]
+        # Get good states
+        good_states = []
+        for b in bin_str:
+            for e_idx, e_val in e2idx:
+                b = b[:e_idx] + str(e_val) + b[e_idx:]
+            good_states.append(b)
+        # Get statevector by transform good states w.r.t its index to 1 and o/w to 0
+        oracle = Statevector(
+            [int(format(i, f"0{num_qubits}b") in good_states) for i in range(2**num_qubits)]
+        )
+        return GroverOperator(oracle, state_preparation=self._circ)
+
+    def _run_circuit(self, circuit: QuantumCircuit) -> Dict[str, float]:
+        """Run the quantum circuit for the number of shots on the Aer simulator backend."""
+        # Sample from circuit
+        job = self.sampler.run(circuit, shots=self.shots)
```

Comment:
My take would when the user specifies the Sampler, they configure it, with shots or whatever as they see fit. It was expected that the primitives might eventually have different options for the outcome e.g. some error definition where shots was tailored to achieve that. Such a primitive could ignore or throw an error on such a call explicitly configured with shots. I.e I would recommend remove explicit shots from init and just work with whatever Sampler is passed - this is what is done elsewhere in algorithms here in general.

Reply:
Make sense. I will do so.